ArduPlane/radio.cpp

```diff
@@ -259,7 +259,21 @@ void Plane::control_failsafe()
         channel_roll->set_control_in(0);
         channel_pitch->set_control_in(0);
         channel_rudder->set_control_in(0);
-        channel_throttle->set_control_in(0);
+
+        switch (control_mode) {
+            case QSTABILIZE:
+            case QHOVER:
+            case QLOITER:
+            case QLAND: // throttle is ignored, but reset anyways
+            case QRTL:  // throttle is ignored, but reset anyways
+            case QAUTOTUNE:
+                // set half throttle to avoid descending at maximum rate
+                channel_throttle->set_control_in(channel_throttle->get_range() / 2);
```

Comment:
instead of div2, wouldn't using get_throttle_hover() be better?

Reply:
No, this is the input side, so we actually want 50% of the stick value, not the throttle hover. If you used throttle hover this can get you a very low range input which would still cause a decent.

Reply:
don't you mean *0.5f   :)

Reply:
uh no? It's a uint, and I want a uint, so why convert to/from float and back? Trust the compiler sometimes. This is actually a bit shift right which is better....